incentivization.md

```diff
@@ -52,231 +47,186 @@ Early P2P file-sharing networks employed reputation-based approaches and sticky
 
-With that in mind, we suggest the scheme where the client pays first.
+With that in mind, we suggest that the client pays first.
 It is simpler than splitting the payment, which would involve a) two payments, and b) negotiating the split.
-It is also simpler than a trusted third party (the centralized flavor of which we want to avoid anyway).
+It is also simpler than a trusted third party (the centralized flavor of which we want to avoid).
+
 Comparing to "client pays after the fact", we observe that there is a balance between risk and privacy.
-If the server "pays first", it assumes risk.
-This risk should be decreased or paid for.
-Decreasing the risk means keeping track of the clients' reputation from the server's standpoint, which may endanger clients' privacy.
-Paying for the risk means increasing prices (i.e., well-behaved clients in aggregate pay for free-riders).
-We suggest that the preferable design is the opposite: the client assumes the risk.
-Why this is better:
-- it's more likely that the server is professionalized: serving data is its business which it wouldn't want to sabotage;
-- the client keeps their privacy, essentially paying for privacy with taking on more risk - this is OK, as risk is "anonymous", and reputation is not.
+If the server "pays first", it assumes risk, which should be decreased or paid for.
+Decreasing the risk means that the client keeps track of the clients' reputation, which endangers privacy.
+Paying for the risk means higher prices (well-behaved clients pay for free-riders).
 
-### Reputation accounting
+We propose that the client assumes the risk and pays for it because:
+- the server is more likely to be professionalized, so dropping paid requests would sabotage its business;
+- the client pays for their privacy by assuming risk, which is acceptable (risk is "anonymous", reputation is not).
 
-Our protocol assumes that the client trusts the server.
-In particular, the client pays first, and then hopes that the server sends back the response.
-A server may technically take the money and do nothing.
-To discourage this behavior, we use reputation: a client keeps track of the server's behavior.
+### Reputation accounting
 
-The MVP version could be:
-- all servers start with zero reputation
+We use reputation to discourage the server from taking the payment and not responding.
+The client keeps track of the server's reputation:
+- all servers start with zero reputation;
 - if the server honors the request, it gets +1;
-- if the server does not respond after the initial query, it gets -1;
-- if the server takes the money and _then_ does not respond, it gets banned (this client will never query it again).
+- if the server does not respond to the initial request, it gets -1;
+- if the server takes the money and does not respond before a timeout, the client will never query it again.
```

Comment:
There seems to be no reflection of the period for which the client expects the message to be available. Identical messages stored for different periods (or is there a default period that applies?) with the same number of requests will yield the same reputation with different costs. 

Does the server get +1 for every request or only the initial one? What is the motivation for limiting -1 only for initial request? 

Can other clients/entities observe such client-server interaction or query this message themselves? Is the server's reputation only local to the particular client or collected by everyone watching it?

Reply:
> the period for which the client expects the message to be available

For MVP, there could be a network-wide constant for a default storage time (see also https://github.com/waku-org/research/issues/39).

> limiting -1 only for initial request

Imprecise wording on my part: "initial request" here just means "request" (to emphasize that this happens before the payment).

> Can other clients/entities observe such client-server interaction

I assume not, the interaction is point-to-point AFAIU.

> or query this message themselves?

They can, why not?

> Is the server's reputation only local to the particular client or collected by everyone watching it?

For now, I assume reputation is local. Global reputation is a) difficult to implement without consensus; b) can lead to censorship.

Reply:
Makes sense, thanks.

> They can, why not?
Just wondering if I, as a client not involved in a particular server-client interaction, can do the following: Suppose that I know a subset of other clients' messages stored with the server (collected privately or by observing data provided by the server previously). Then, I can query these messages from the server (periodically) over time to assess its reputation. This results in substantial dummy traffic generated with the sole purpose of getting a solid local reputation scoring.

Reply:
@s-tikhomirov are you familiar with Eigentrust or Web of Trust approaches? Could these perhaps provide a balance between only local reputation and requiring the overhead for global state consensus?

Reply:
> a subset of other clients' messages stored with the server (collected privately or by observing data provided by the server previously)

I'm not sure I understand what "collected privately" means here. In my understanding, all servers serve (subsets of) the same dataset - all previously broadcast messages. In any case though, the problem of dummy traffic may arise.

It is quite similar to probing in the Lightning Network (I wrote a [paper](https://eprint.iacr.org/2021/384) on that): to increase payment reliability, LN nodes send dummy failing payments to estimate balances in remote channels. Such dummy payments constitute the majority of traffic (at least they did ~2 years ago), as reported by big node operators.

Reply:
> are you familiar with Eigentrust or Web of Trust approaches?

Eigentrust - no, Web of Trust - somewhat. I'm going to have a deeper look, thank you for the pointers.

Reply:
Created an issue about future research on existing reputation systems (#47), closing this thread.